apps/internal/base/base.go

```diff
@@ -302,3 +306,30 @@ func convertStrUnixToUTCTime(unixTime string) (time.Time, error) {
 	}
 	return time.Unix(timeInt, 0).UTC(), nil
 }
+
+func getKeyFromRequest(authParams authority.AuthParams) string {
+	if authParams.AuthorizationType == authority.ATClientCredentials {
+		return getAppKey(authParams)
+	}
+	if authParams.AuthorizationType == authority.ATRefreshToken {
+		return authParams.HomeaccountID
+	}
+	return ""
+}
+
+func getKeyFromResponse(authParams authority.AuthParams, homeaccountIDfromResponse string) string {
```

Comment:
So, want to note a few trends I see.  

Java names:  homeaccountIDfromResponse, homeAccountID would be fine.  
getKeyFromResponse: There are rare times you should use "get" in a name, and that is almost always because you have a name conflict.  keyFromResponse() says the same thing.  keyFromResp() is even better.

I'd think about attaching this to accesstokens.TokenResponse as CacheKey(authParams authority.AuthParams).

Reply:
Resoved in https://github.com/AzureAD/microsoft-authentication-library-for-go/pull/184/commits/71dea81ddef28df17b4741bef3e8643d75e7ccd9